commerce_coordinator/apps/titan/clients.py

```diff
@@ -208,25 +208,36 @@ def get_payment(self, edx_lms_user_id=None, payment_number=None):
 
         return response['data']['attributes']
 
-    def update_payment(self, payment_number, payment_state, response_code):
+    def update_payment(
+        self, edx_lms_user_id, order_uuid, payment_number, payment_state, reference_number, provider_response_body=None
+    ):
         """
         Request Titan to update payment.
 
         Args:
+            edx_lms_user_id: The edx.org LMS user ID of the user making the payment.
+            order_uuid: Order UUID related to this order.
             payment_number: The Payment identifier in Spree.
             payment_state: State to advance the payment to.
-            response_code: Payment attempt response code provided by stripe.
+            reference_number: Payment client intent key.
+            provider_response_body: Optional. The saved response from a request to the payment provider.
         """
+        data_attributes = {
+            'edxLmsUserId': edx_lms_user_id,
+            'orderUuid': order_uuid,
+            'paymentNumber': payment_number,
+            'paymentState': payment_state,
+            'referenceNumber': reference_number,
```

Comment:
Renamed to `referenceNumber` from `responseCode`. Discovered this while implementation that this has been renamed in our API spec.